partiql-ast/src/main/java/org/partiql/ast/v1/sql/SqlDialect.kt

```diff
@@ -0,0 +1,824 @@
+        } else {
+            var t = tail
+            t = t concat node.symbol + "("
+            t = visitExprWrapped(node.rhs, t)
+            t = t concat ")"
+            return t
+        }
+    }
+
+    override fun visitExprAnd(node: ExprAnd, tail: SqlBlock): SqlBlock {
+        var t = tail
+        t = visitExprWrapped(node.lhs, t)
+        t = t concat " AND "
+        t = visitExprWrapped(node.rhs, t)
+        return t
+    }
+
+    override fun visitExprOr(node: ExprOr, tail: SqlBlock): SqlBlock {
+        var t = tail
+        t = visitExprWrapped(node.lhs, t)
+        t = t concat " OR "
+        t = visitExprWrapped(node.rhs, t)
+        return t
+    }
+
+    override fun visitExprNot(node: ExprNot, tail: SqlBlock): SqlBlock {
+        var t = tail
+        t = t concat "NOT ("
+        t = visitExprWrapped(node.value, t)
+        t = t concat ")"
+        return t
+    }
+
+    override fun visitExprVarRef(node: ExprVarRef, tail: SqlBlock): SqlBlock {
+        var t = tail
+        // Prepend @
+        if (node.scope.code() == Scope.LOCAL) {
+            t = t concat "@"
+        }
+        t = visitIdentifierChain(node.identifierChain, t)
+        return t
+    }
+
+    override fun visitExprSessionAttribute(node: ExprSessionAttribute, tail: SqlBlock): SqlBlock =
+        tail concat node.sessionAttribute.name()
+
+    override fun visitExprPath(node: ExprPath, tail: SqlBlock): SqlBlock {
+        var t = visitExprWrapped(node.root, tail)
+        var cur = node.next
+        while (cur != null) {
+            t = visitPathStep(cur, t)
+            cur = cur.next
+        }
+        return t
+    }
+
+    override fun visitPathStepField(node: PathStep.Field, tail: SqlBlock): SqlBlock =
+        tail concat ".${node.field.sql()}"
+
+    override fun visitPathStepElement(node: PathStep.Element, tail: SqlBlock): SqlBlock {
+        var t = tail
+        val key = node.element
+        // use [ ] syntax
+        t = t concat "["
+        t = visitExprWrapped(key, t)
+        t = t concat "]"
+        return t
+    }
+
+    override fun visitPathStepAllElements(node: PathStep.AllElements, tail: SqlBlock): SqlBlock = tail concat "[*]"
+
+    override fun visitPathStepAllFields(node: PathStep.AllFields, tail: SqlBlock): SqlBlock = tail concat ".*"
+
+    @OptIn(PartiQLValueExperimental::class)
+    override fun visitExprCall(node: ExprCall, tail: SqlBlock): SqlBlock {
+        var t = tail
+        val f = node.function
+        // Special case -- COUNT() maps to COUNT(*)
+        if (f.next == null && f.root.symbol.uppercase() == "COUNT" && node.args.isEmpty()) {
+            return t concat "COUNT(*)"
+        }
+        // Special case -- DATE_ADD('<datetime_field>', <lhs>, <rhs>) -> DATE_ADD(<datetime_field>, <lhs>, <rhs>)
+        // Special case -- DATE_DIFF('<datetime_field>', <lhs>, <rhs>) -> DATE_DIFF(<datetime_field>, <lhs>, <rhs>)
+        // TODO this will need to be rewritten once PartiQLValue is removed from AST
```

Comment:
(self-review) special conversion for `DATE_ADD` and `DATE_DIFF` since in the v1 AST, we model them as `ExprCall`s rather than dedicated nodes.